airbrakes/hardware/imu.py

```diff
@@ -64,99 +70,117 @@ def __init__(self, port: str) -> None:
-            imu_data_packet.invalid_fields.append(channel)
-
-    @staticmethod
-    def _process_packet_data(packet: "mscl.MipDataPacket", imu_data_packet: IMUDataPacket) -> None:
-        """
-        Process the data points in the packet and update the data packet object. Modifies
-        `imu_data_packet` in place.
-        :param packet: The IMU data packet containing multiple data points.
-        :param imu_data_packet: The initialized data packet object to populate.
-        """
-        # Iterate through each data point in the packet.
-        data_point: mscl.MipDataPoint
-        for data_point in packet.data():
-            # Extract the channel name of the data point.
-            channel = data_point.channelName()
-
-            # Check if the channel is relevant for the data packet, we check for quaternions
-            # separately because the IMU sends them over as a matrix, but we store each of them
-            # individually as fields.
-            if hasattr(imu_data_packet, channel) or "Quaternion" in channel:
-                # Process and set the data point value in the data packet.
-                IMU._process_data_point(data_point, channel, imu_data_packet)
-
-    def _fetch_data_loop(self, port: str) -> None:
+    def _fetch_data_loop(self, port: str) -> None:  # pragma: no cover
         """
         Continuously fetch data packets from the IMU and process them.
         :param port: The serial port to connect to the IMU.
         """
+        # Set the process priority really high, as we want to get the data from the IMU as fast as
+        # possible:
+        set_process_priority(-15)
+
         # Connect to the IMU and initialize the node used for getting data packets
         connection = mscl.Connection.Serial(port)
         node = mscl.InertialNode(connection)
+        packet: mscl.MipDataPacket
+        data_point: mscl.MipDataPoint
 
         while self.is_running:
             # Retrieve data packets from the IMU.
             packets: mscl.MipDataPackets = node.getDataPackets(timeout=10)
 
-            packet: mscl.MipDataPacket
+            self._fetched_imu_packets.value = len(packets)
+
+            messages = []
             for packet in packets:
-                # Initialize the appropriate data packet.
-                imu_data_packet = IMU._initialize_packet(packet)
-                if imu_data_packet is None:
-                    # Skip unrecognized packets.
+                # Extract the timestamp from the packet.
+                timestamp = packet.collectedTimestamp().nanoseconds()
+
+                # Initialize packet with the timestamp, determines if the packet is raw or estimated
+                if packet.descriptorSet() == ESTIMATED_DESCRIPTOR_SET:
+                    imu_data_packet = EstimatedDataPacket(timestamp)
+                elif packet.descriptorSet() == RAW_DESCRIPTOR_SET:
+                    imu_data_packet = RawDataPacket(timestamp)
```

Comment:
```suggestion
                if packet.descriptorSet() == RAW_DESCRIPTOR_SET:
                    imu_data_packet = RawDataPacket(timestamp)
                elif packet.descriptorSet() == ESTIMATED_DESCRIPTOR_SET:
                    imu_data_packet = EstimatedDataPacket(timestamp)
```
micro-optimization: since raw freq is 2x est, this shortens the path the code has to take? Not sure if the newer python versions bytecode already optimize for this but worth doing anyway